packages/backend/src/open_payments/connection/middleware.ts

```diff
@@ -0,0 +1,23 @@
+import { AppContext } from '../../app'
+import { IncomingPayment } from '../payment/incoming/model'
+
+export interface ConnectionContext extends AppContext {
+  incomingPayment: IncomingPayment
+}
+
+export const connectionMiddleware = async (
```

Comment:
🤔 
since we would only need this if we are dealing with `application/spsp4+json`, maybe if we had the SPSP middleware in a function (ie the comment above), we could just do this incoming payment lookup directly in that method, within the `if (application/spsp4+json)` block?

Reply:
`ctx.incomingPayment` is also used in the (non-SPSP) `ConnectionRoutes.get`